distributed/scheduler.py

```diff
@@ -2315,19 +2327,49 @@ def decide_worker(self, ts: TaskState) -> WorkerState:
         """
         Decide on a worker for task *ts*.  Return a WorkerState.
         """
+        if not self._workers_dv:
+            return None
+
         ws: WorkerState = None
+        group: TaskGroup = ts._group
         valid_workers: set = self.valid_workers(ts)
 
         if (
             valid_workers is not None
             and not valid_workers
             and not ts._loose_restrictions
-            and self._workers_dv
         ):
             self._unrunnable.add(ts)
             ts.state = "no-worker"
             return ws
 
+        # Group is larger than cluster with few dependencies? Minimize future data transfers.
+        if (
+            valid_workers is None
```

Comment:
IIUC as soon as I am using any kind of restrictions, even if they are loose (`allow_other_workers==True`), the `valid_workers` will be non-null. Users would see vastly different scheduling decisions and will likely have a very hard time understanding why.

Reply:
@gjoseph92 used to do this, I encouraged him to drop it.  It made the code significantly more complex.  I'm inclined to start with what we have here, merge it in, and address worker restrictions in the future if they come up.